release/models/platform/openconfig-platform.yang

```diff
@@ -534,6 +541,89 @@ module openconfig-platform {
     }
   }
 
+  grouping platform-health-top {
+    description
+      "Grouping containing health-related parameters.";
+
+    container health {
+      description
+        "The health of the component. The paramaters within this
+        container indicate the status of the component beyond whether
+        it is operationally up or down. When a signal is received
+        that a component is in an unhealth state the gNOI.Healthz
+        service can be used to retrieve further diagnostic information
+        relating to the component.
+
+        The contents of this directory relate only to the specific
+        component that it is associated with. In the case that child
+        components become unhealthy and this causes a parent component
+        to be unhealthy, the new unhealthy status should be reported at
+        both components, such that an interested system can take the
+        relevant actions (e.g., retrieve the Healthz output, or
+        apply mitigation actions).";
+      reference
+        "https://https://github.com/openconfig/gnoi/tree/master/healthz";
+
+      container state {
+        config false;
+        description
+          "Operational state parameters relating to component health.";
+        uses platform-health-state;
+      }
+    }
+  }
+
+  grouping platform-health-state {
+    description
+      "Operational state parmaeters relating to a platform component's
+      health.";
+
+    leaf status {
+      type enumeration {
+        enum HEALTHY {
+          description
+            "The component is in a HEALTHY state, and is operating
+            within the expected parameters.";
+        }
+        enum DEGRADED {
+          description
+            "The component is in a degraded state. That is to say that
+            it has not completely failed, but failures have occurred that
+            either reduce its ability to perform, or may cause it to reach
+            an unhealthy state should further failures occur.
+
+            An example of a degraded state may be a fan tray which has only
+            a subset of the fans it contains having failed.";
+        }
+        enum FAILED {
```

Comment:
These do not align with gNOI healthz, probably they should?  

https://github.com/openconfig/gnoi/blob/3e9a99fe151c1801b9fe2fbd7b68ca0a066396fc/healthz/healthz.proto#L53-L57

Reply:
@robshakir - reference https://github.com/openconfig/gnoi/pull/65 as well as this is directly related

Reply:
@marcushines @robshakir 

The current gNOI healthz proto has the following enum definition
```
enum Status {
  STATUS_UNSPECIFIED = 0;   // The path doesn't support healthz
  STATUS_HEALTHLY = 1;      // The path is healthy
  STATUS_UNHEALTHY = 2;     // The path is unhealthy
}
```
Whereas the YANG enum mapping is as follows (since values are not specified)
```
HEALTHY = 0
DEGRADED = 1
FAILED = 2
```
I would suggest alignment here at a minimum and may want to consider tagging values in the YANG enum definition such as:

```
UNSPECIFIED = 0
HEALTHY = 1
DEGRADED = 2
FAILED = 3
```

Reply:
i think the status should align - as far as the value of degraded - if that in fact means to say - hey i am probably aggregating other things - so degraded means i am good but some of my subcomponents are not ... meh not sure i see the value - basically it becomes a very noisy signal for probably not much good

i.e. if a line card defines it's health around interface status (which it shouldn't) then it would be in a degraded state likely often but that actually doesn't reflect anything of value really

now if the linecard defines it's status around optic health - and there is an installed optic which is not reporting - that status degraded would make sense...

but again it wouldn't tell you anything about the actual linecard
vs say a switch fabric being off line and that linecard actually not having full capacity ... that would make alot of sense for degraded state

Reply:
I think given the ambiguity of `DEGRADED` I'll remove it. I haven't added `UNSPECIFIED` as generally we include this in the protobuf to handle the case that the `proto3` field is not set, and YANG doesn't have this implicit default value. Code generation creates it where required for the language.